src/test/java/bio/terra/integration/AzureIntegrationTest.java

```diff
@@ -474,6 +474,41 @@ sourceFileAzure, getSourceStorageAccountPrimarySharedKey()))
     DatasetSpecificationModel datasetSchema = datasetModel.getSchema();
 
     // dataset ingest
+    // Ingest Metadata - 1 row in ARRAY mode
+    String arrayIngestTableName = "person";
+    String personIdField = "person_id";
+    String personYearOfBirthField = "year_of_birth";
+    Map<String, Integer> records = Map.of(personIdField, 1, personYearOfBirthField, 1980);
+    IngestRequestModel arrayIngestRequest =
+        new IngestRequestModel()
+            .ignoreUnknownValues(false)
+            .maxBadRecords(0)
+            .table(arrayIngestTableName)
+            .profileId(profileId)
+            .addRecordsItem(records)
+            .format(IngestRequestModel.FormatEnum.ARRAY)
+            .loadTag(Names.randomizeName("azureArrayIngest"));
+
+    IngestResponseModel arrayIngestResponse =
+        dataRepoFixtures.ingestJsonData(steward, datasetId, arrayIngestRequest);
+    assertThat("1 row was ingested", arrayIngestResponse.getRowCount(), equalTo(1L));
+    tableRowCount.put(arrayIngestTableName, 1);
+
+    Object firstPersonRow =
+        dataRepoFixtures
+            .retrieveDatasetData(steward, datasetId, arrayIngestTableName, 0, 1, null)
+            .getResult()
+            .get(0);
+    assertThat(
+        "record looks as expected - person_id",
+        ((LinkedHashMap) firstPersonRow).get(personIdField),
+        equalTo(records.get(personIdField)));
+
+    assertThat(
+        "record looks as expected - person_id",
+        ((LinkedHashMap) firstPersonRow).get(personYearOfBirthField),
+        equalTo(records.get(personYearOfBirthField)));
```

Comment:
This is following the convention of the other code here but I think this can be improved.

- using the type `LinkedHashMap` seems too specific here; would `Map` work instead? That would avoid making this code break if the underlying implementation changes.
- the code casts `firstPersonRow` twice but it's the same cast each time. Casting once at the declaration of `firstPersonRow` would be better.
- the code is doing an unsafe cast but doesn't use `SuppressWarnings`. I don't know if it's worth changing the existing tests but it seems like we should write new code with `SuppressWarnings`.
- there are hamcrest matchers for these checks that should produce better messages if the test fails.
I didn't try to run it, but this could work:
```
    @SuppressWarnings("unchecked")
    Map<Object, Object> firstPersonRow =
        (Map<Object, Object>) dataRepoFixtures
            .retrieveDatasetData(steward, datasetId, arrayIngestTableName, 0, 1, null)
            .getResult()
            .get(0);

    assertThat(firstPersonRow, allOf(
        hasEntry(personIdField, records.get(personIdField)),
        hasEntry(personYearOfBirthField, records.get(personYearOfBirthField))));
```

Reply:
I actually wonder if just making the `retrieveDatasetData` fixture method return an object where records is a `List<Map<String, Object>>` as opposed to an object?  I think then using it result in tests would be much simpler